runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -258,13 +259,38 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 			type -= MENU_ACTION_DEPRIORITIZE_OFFSET;
 		}
 
-		if (config.highlightMenuNames() &&
-			NPC_MENU_ACTIONS.contains(MenuAction.of(type)) &&
-			highlightedNpcs.stream().anyMatch(npc -> npc.getIndex() == event.getIdentifier() && (!npc.isDead() || config.highlightDeadNpcs())))
+		if (NPC_MENU_ACTIONS.contains(MenuAction.of(type)))
 		{
+			Color menuColor = null;
+			final NPC menuNpc = client.getCachedNPCs()[event.getIdentifier()];
+			final boolean highlightMenuNpc = config.highlightMenuNames() && highlightedNpcs.contains(menuNpc);
+			if (menuNpc.isDead())
+			{
+				if (config.getdeadNpcColor() != null)
+				{
+					menuColor = config.getdeadNpcColor();
+				}
+				else if (config.highlightDeadNpcs() && highlightMenuNpc)
```

Comment:
Why is there both an option to highlight dead npcs *and* an npc highlight color that appear separate? Eg you can have deadnpccolor -> red & highlightdeadnpcs -> off - and it would highlight dead npcs red.